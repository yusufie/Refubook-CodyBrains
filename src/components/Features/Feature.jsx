import React from 'react';
import star from '../../assets/star.svg';

const Feature = ({ item }) => {
  return (
    <div className=" max-w-sm flex flex-col items-center   gap-4 py-5">
      <div className="flex items-center justify-start w-full gap-4">
        <img src={star} alt="" />
        <h4 className="text-lg text-refubookLightBlack text-left font-bold">
          {item?.title}
        </h4>
      </div>
      <p className="text-md text-refubookLightBlack font-medium text-left">
        {item?.text}
      </p>
    </div>
  );
};

export default Feature;
