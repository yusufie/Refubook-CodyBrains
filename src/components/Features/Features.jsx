import { Link } from 'react-router-dom';
import { Swiper, SwiperSlide } from 'swiper/react';
import { Pagination } from 'swiper';
import { useTranslation } from 'react-i18next';
// import featuresData from '../../data/featuresData';
import i18next from 'i18next';
import Feature from './Feature';
import illustration from '../../assets/features-illustration.svg';
import star from '../../assets/star.svg';

// The Features component is a functional component that displays information about RefuBook.
const Features = () => {
  const { t } = useTranslation();

  const featuresData = [
    {
      id: 1,
      title: t('Reliability'),
      text: t('Reliability_Text'),
    },
    {
      id: 2,
      title: t('Motivation'),
      text: t('Motivation_Text'),
    },
    {
      id: 3,
      title: t('Efficiency'),
      text: t('Efficiency_Text'),
    },
    {
      id: 4,
      title: t('Creativity'),
      text: t('Creativity_Text'),
    },
  ];

  return (
    // The component consists of a div with a blue background and two main sections: a section with text and a button, and a section with an illustration and a list of features.
    <div className="w-full flex flex-col items-center justify-center bg-refubookAboutBlue py-4 mt-5">
      <div className=" w-3/4  md:w-2/3 flex flex-col items-center gap-4 justify-center py-5">
        {/* The first section contains a heading and a paragraph of text. */}
        <h1 className="text-4xl md:text-3xl xl:text-4xl   text-refubookBlue font-bold text-center">
          {t('About')}
        </h1>
        <p className="text-md leading-relaxed md:text-lg  text-refubookLightBlack font-medium text-justify md:text-center">
          {t('About_RefuBook_Website')}
        </p>
        {/* The first section also contains a button with a link to the "about" page. */}
        <Link to="about">
          <button
            type="button"
            className=" py-1 px-3 md:py-2 md:px-9 bg-refubookBlue text-refubookWhite font-bold text-lg md:text-xl  rounded-full shadow-lg hover:bg-refubookWhite hover:text-refubookBlack transition duration-500 ease-in-out"
          >
            {t('Learn_More')}
          </button>
        </Link>
      </div>
      <div className="w-full flex flex-col items-center md:flex-row-reverse justify-center 2xl:justify-evenly gap-4 py-5">
        {/* The second section contains an illustration and a list of features. */}
        <img src={illustration} alt="" />
        {/* The Slider component is used to display the list of features on smaller screens. */}
        <div className="block md:hidden w-full max-w-sm">
          {/*   The Swiper component is wrapped around the content that we want to display as slides.
    We pass in the pagination prop and set clickable to true to enable pagination for the Swiper.
    We also pass in the Pagination module to enable pagination functionality. */}
          <Swiper
            pagination={{
              clickable: true,
            }}
            modules={[Pagination]}
          >
            {/* We map over the featuresData array to render a SwiperSlide for each item in the array. */}
            {featuresData?.map((item) => {
              return (
                // The SwiperSlide component is used to wrap the content that we want to display as a slide.
                // We pass a unique key prop to each SwiperSlide to help with rendering performance.
                <SwiperSlide key={item.id}>
                  {/* The content for each slide is a div containing an image, a title, and some text. */}
                  <div className=" flex flex-col items-center justify-center gap-5 py-5">
                    <img src={star} alt="" />
                    <p className="text-md  text-refubookLightBlack font-medium text-center">
                      {i18next.t(item.title)}
                    </p>
                    <p className="text-md  text-refubookLightBlack font-medium text-center mb-4">
                      {i18next.t(item.text)}
                    </p>
                  </div>
                </SwiperSlide>
              );
            })}
          </Swiper>
        </div>
        {/* On larger screens, the features are displayed in a grid using the Feature component. */}
        <div className="hidden md:grid grid-cols-2 grid-rows-2 gap-2 justify-start px-10">
          {/* We map over the featuresData array to render a Feature component for each item in the array. */}
          {featuresData?.map((item) => {
            return <Feature key={item.id} item={item} />;
          })}
        </div>
      </div>
    </div>
  );
};

export default Features;
