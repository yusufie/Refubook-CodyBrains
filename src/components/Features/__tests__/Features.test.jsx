import renderer from 'react-test-renderer';
import Features from '../Features';

jest.mock('react-router-dom', () => ({
  Link: 'Link',
}));

jest.mock('swiper/react', () => ({
  Swiper: 'Swiper',
  SwiperSlide: 'SwiperSlide',
}));

jest.mock('swiper', () => ({
  Pagination: 'Pagination',
}));

describe('Features', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Features />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
