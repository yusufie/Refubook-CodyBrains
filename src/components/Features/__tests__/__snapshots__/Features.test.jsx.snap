// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features renders correctly 1`] = `
<div
  className="w-full flex flex-col items-center justify-center bg-refubookAboutBlue py-4 mt-5"
>
  <div
    className=" w-3/4  md:w-2/3 flex flex-col items-center gap-4 justify-center py-5"
  >
    <h1
      className="text-4xl md:text-3xl xl:text-4xl   text-refubookBlue font-bold text-center"
    >
      About
    </h1>
    <p
      className="text-md leading-relaxed md:text-lg  text-refubookLightBlack font-medium text-justify md:text-center"
    >
      About_RefuBook_Website
    </p>
    <Link
      to="about"
    >
      <button
        className=" py-1 px-3 md:py-2 md:px-9 bg-refubookBlue text-refubookWhite font-bold text-lg md:text-xl  rounded-full shadow-lg hover:bg-refubookWhite hover:text-refubookBlack transition duration-500 ease-in-out"
        type="button"
      >
        Learn_More
      </button>
    </Link>
  </div>
  <div
    className="w-full flex flex-col items-center md:flex-row-reverse justify-center 2xl:justify-evenly gap-4 py-5"
  >
    <img
      alt=""
      src="features-illustration.svg"
    />
    <div
      className="block md:hidden w-full max-w-sm"
    >
      <Swiper
        modules={
          Array [
            "Pagination",
          ]
        }
        pagination={
          Object {
            "clickable": true,
          }
        }
      >
        <SwiperSlide>
          <div
            className=" flex flex-col items-center justify-center gap-5 py-5"
          >
            <img
              alt=""
              src="star.svg"
            />
            <p
              className="text-md  text-refubookLightBlack font-medium text-center"
            />
            <p
              className="text-md  text-refubookLightBlack font-medium text-center mb-4"
            />
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div
            className=" flex flex-col items-center justify-center gap-5 py-5"
          >
            <img
              alt=""
              src="star.svg"
            />
            <p
              className="text-md  text-refubookLightBlack font-medium text-center"
            />
            <p
              className="text-md  text-refubookLightBlack font-medium text-center mb-4"
            />
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div
            className=" flex flex-col items-center justify-center gap-5 py-5"
          >
            <img
              alt=""
              src="star.svg"
            />
            <p
              className="text-md  text-refubookLightBlack font-medium text-center"
            />
            <p
              className="text-md  text-refubookLightBlack font-medium text-center mb-4"
            />
          </div>
        </SwiperSlide>
        <SwiperSlide>
          <div
            className=" flex flex-col items-center justify-center gap-5 py-5"
          >
            <img
              alt=""
              src="star.svg"
            />
            <p
              className="text-md  text-refubookLightBlack font-medium text-center"
            />
            <p
              className="text-md  text-refubookLightBlack font-medium text-center mb-4"
            />
          </div>
        </SwiperSlide>
      </Swiper>
    </div>
    <div
      className="hidden md:grid grid-cols-2 grid-rows-2 gap-2 justify-start px-10"
    >
      <div
        className=" max-w-sm flex flex-col items-center   gap-4 py-5"
      >
        <div
          className="flex items-center justify-start w-full gap-4"
        >
          <img
            alt=""
            src="star.svg"
          />
          <h4
            className="text-lg text-refubookLightBlack text-left font-bold"
          >
            Reliability
          </h4>
        </div>
        <p
          className="text-md text-refubookLightBlack font-medium text-left"
        >
          Reliability_Text
        </p>
      </div>
      <div
        className=" max-w-sm flex flex-col items-center   gap-4 py-5"
      >
        <div
          className="flex items-center justify-start w-full gap-4"
        >
          <img
            alt=""
            src="star.svg"
          />
          <h4
            className="text-lg text-refubookLightBlack text-left font-bold"
          >
            Motivation
          </h4>
        </div>
        <p
          className="text-md text-refubookLightBlack font-medium text-left"
        >
          Motivation_Text
        </p>
      </div>
      <div
        className=" max-w-sm flex flex-col items-center   gap-4 py-5"
      >
        <div
          className="flex items-center justify-start w-full gap-4"
        >
          <img
            alt=""
            src="star.svg"
          />
          <h4
            className="text-lg text-refubookLightBlack text-left font-bold"
          >
            Efficiency
          </h4>
        </div>
        <p
          className="text-md text-refubookLightBlack font-medium text-left"
        >
          Efficiency_Text
        </p>
      </div>
      <div
        className=" max-w-sm flex flex-col items-center   gap-4 py-5"
      >
        <div
          className="flex items-center justify-start w-full gap-4"
        >
          <img
            alt=""
            src="star.svg"
          />
          <h4
            className="text-lg text-refubookLightBlack text-left font-bold"
          >
            Creativity
          </h4>
        </div>
        <p
          className="text-md text-refubookLightBlack font-medium text-left"
        >
          Creativity_Text
        </p>
      </div>
    </div>
  </div>
</div>
`;
